pytorch_lightning/tuner/lr_finder.py

```diff
@@ -124,10 +128,14 @@ def func(trainer):
             args = (optimizer, self.lr_max, self.num_training)
             scheduler = _LinearLR(*args) if self.mode == "linear" else _ExponentialLR(*args)
             sched_config = _get_default_scheduler_config()
-            sched_config.update({"scheduler": scheduler, "interval": "step"})
+            sched_config.update({"scheduler": scheduler, "interval": "step", "opt_idx": 0})
 
-            trainer.strategy.optimizers = [optimizer]
-            trainer.strategy.lr_schedulers = [sched_config]
+            optimizers = [optimizer]
+            lr_schedulers = [sched_config]
+            _attach_scheduler_opt_idx(optimizers, lr_schedulers)
+
+            trainer.strategy.optimizers = optimizers
+            trainer.strategy.lr_schedulers = lr_schedulers
```

Comment:
Keep the existing structure?

```suggestion
            trainer.strategy.optimizers = [optimizer]
            trainer.strategy.lr_schedulers = [sched_config]
            _attach_scheduler_opt_idx(trainer.strategy.optimizers, trainer.strategy.lr_schedulers)
```